mobile/mobile-core/android-process-service.ts

```diff
@@ -0,0 +1,189 @@
+///<reference path="./../../.d.ts"/>
+"use strict";
+
+import {EOL} from "os";
+import * as deviceAndroidDebugBridgePath from "../android/device-android-debug-bridge";
+
+export class AndroidProcessService implements Mobile.IAndroidProcessService {
+	// Describes one row from the adb shell cat proc/net/tcp result.
+	private static ANDROID_PORT_INFORMATION_REGEX = /([0-9]+):\W+([0-9A-Za-z]+:[0-9A-Za-z]+)\W+([0-9A-Za-z]+:[0-9A-Za-z]+)\W+([0-9A-Za-z]+)\W+[0-9A-Za-z]+:[0-9A-Za-z]+\W+[0-9A-Za-z]+:[0-9A-Za-z]+ +[0-9A-Za-z]+\W+([0-9]+)/g;
+
+	constructor(private $errors: IErrors,
+		private $staticConfig: Config.IStaticConfig,
+		private $injector: IInjector) { }
+
+	public mapAbstractToTcpPort(deviceIdentifier: string, appIdentifier: string): IFuture<string> {
+		return (() => {
+			let adb = this.getAdb(deviceIdentifier);
+			let processId = this.getProcessId(adb, appIdentifier).wait();
+			if (!processId) {
+				this.$errors.failWithoutHelp(`The application is not started on the device with identifier ${deviceIdentifier}.`);
+			}
+
+			let abstractPort = this.getAbstractPortForApplication(adb, processId, appIdentifier).wait();
+
+			if (!abstractPort) {
+				this.$errors.failWithoutHelp(`The application is not started on the device with identifier ${deviceIdentifier}.`);
```

Comment:
you can extract the error message in a variable, as you are using it on two places